rust/src/lib/unit_tests/route.rs

```diff
@@ -864,3 +864,155 @@ fn test_route_advmss_deserialize_from_string() {
+    let cur_routes: Routes = serde_yaml::from_str(
+        r"
+        config:
+        - destination: 0.0.0.0/0
+          next-hop-interface: eth1
+          next-hop-address: 192.0.2.1
+          table-id: 200
+        - destination: ::/0
+          next-hop-interface: eth1
+          next-hop-address: 2001:db8:1::3
+          table-id: 200
+        ",
+    )
+    .unwrap();
+
+    let des_routes: Routes = serde_yaml::from_str(
+        r"
+        config:
+        - destination: 0.0.0.0/0
+          next-hop-address: 192.0.2.2
+          next-hop-interface: eth1
+          table-id: 200
+        - destination: ::/0
+          next-hop-address: 2001:db8:1::2
+          next-hop-interface: eth1
+          table-id: 200
+        ",
+    )
+    .unwrap();
+
+    let (merged_ifaces, _current_ifaces) = gen_merged_ifaces_for_route_test();
+    let merged_err =
+        MergedRoutes::new(des_routes, cur_routes, &merged_ifaces).unwrap_err();
+
+    assert_eq!(merged_err.kind(), ErrorKind::InvalidArgument);
+    assert!(merged_err.msg().contains("conflicting nexthop"));
+}
+
+#[test]
+fn test_explicit_route_nexthop_change() {
+    let cur_routes: Routes = serde_yaml::from_str(
+        r"
+        config:
+        - destination: 0.0.0.0/0
+          next-hop-interface: eth1
+          next-hop-address: 192.0.2.1
+          table-id: 200
+        - destination: ::/0
+          next-hop-interface: eth2
+          next-hop-address: 2001:db8:1::3
+          table-id: 200
+        ",
+    )
+    .unwrap();
+
+    let des_routes: Routes = serde_yaml::from_str(
+        r"
+        config:
+        - destination: 0.0.0.0/0
+          next-hop-interface: eth1
+          next-hop-address: 192.0.2.1
+          table-id: 200
+          state: absent
+        - destination: 0.0.0.0/0
+          next-hop-address: 192.0.2.2
+          next-hop-interface: eth1
+          table-id: 200
+        - destination: ::/0
+          next-hop-interface: eth2
+          next-hop-address: 2001:db8:1::3
+          table-id: 200
+          state: absent
+        - destination: ::/0
+          next-hop-address: 2001:db8:1::2
+          next-hop-interface: eth2
+          table-id: 200
+        ",
+    )
+    .unwrap();
+
+    let (merged_ifaces, _current_ifaces) = gen_merged_ifaces_for_route_test();
+    let merged_routes =
+        MergedRoutes::new(des_routes, cur_routes, &merged_ifaces).unwrap();
+
+    assert_eq!(
+        merged_routes.merged.get("eth1").unwrap()[0].destination,
+        Some("0.0.0.0/0".to_string())
+    );
+    assert_eq!(
+        merged_routes.merged.get("eth1").unwrap()[0].next_hop_addr,
+        Some("192.0.2.2".to_string())
+    );
+
+    assert_eq!(
+        merged_routes.merged.get("eth2").unwrap()[0].destination,
+        Some("::/0".to_string())
+    );
+    assert_eq!(
+        merged_routes.merged.get("eth2").unwrap()[0].next_hop_addr,
+        Some("2001:db8:1::2".to_string())
+    );
+}
+
+#[test]
+fn test_nexthop_change_absent_routes() {
```

Comment:
Good test! Please a concise comment explaining what is this test guarding against. It's obvious for us now, but we'll forget in the future.